tests/test_utils/gcp_system_decorator.py

```diff
@@ -0,0 +1,141 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import os
+import subprocess
+import unittest
+
+from tests.gcp.utils.gcp_authenticator import GcpAuthenticator
+from tests.contrib.utils.run_once_decorator import run_once
+
+AIRFLOW_MAIN_FOLDER = os.path.realpath(
+    os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), os.pardir, os.pardir, os.pardir
+    )
+)
+
+AIRFLOW_PARENT_FOLDER = os.path.realpath(
+    os.path.join(AIRFLOW_MAIN_FOLDER, os.pardir, os.pardir, os.pardir)
+)
+ENV_FILE_RETRIEVER = os.path.join(
+    AIRFLOW_PARENT_FOLDER, "get_system_test_environment_variables.py"
+)
+
+AIRFLOW_HOME = os.environ.get(
+    "AIRFLOW_HOME", os.path.join(os.path.expanduser("~"), "airflow")
+)
+
+POSTGRES_LOCAL_EXECUTOR = os.path.realpath(
+    os.path.join(
+        AIRFLOW_HOME, "tests", "contrib", "operators", "postgres_local_executor.cfg"
+    )
+)
+
+
+SKIP_TEST_WARNING = """
+The test is only run when the test is run in environment with GCP-system-tests enabled
+environment. You can enable it in one of two ways:
+
+* Set GCP_CONFIG_DIR environment variable to point to the GCP configuration
+  directory which keeps variables.env file with environment variables to set
+  and keys directory which keeps service account keys in .json format
+* Run this test within automated environment variable workspace where
+  config directory is checked out next to the airflow one.
+
+"""
+
+SKIP_LONG_TEST_WARNING = """
+The test is only run when the test is run in with GCP-system-tests enabled
+environment. And environment variable GCP_ENABLE_LONG_TESTS is set to True.
+You can enable it in one of two ways:
+
+* Set GCP_CONFIG_DIR environment variable to point to the GCP configuration
+  directory which keeps variables.env file with environment variables to set
+  and keys directory which keeps service account keys in .json format and
+  set GCP_ENABLE_LONG_TESTS to True
+* Run this test within automated environment variable workspace where
+  config directory is checked out next to the airflow one.
+"""
+
+
+LOCAL_EXECUTOR_WARNING = """
+The test requires local executor. Please set AIRFLOW_CONFIG variable to '{}'
+and make sure you have a Postgres server running locally and
+airflow/airflow.db database created.
+
+You can create the database via these commands:
+'createuser root'
+'createdb airflow/airflow.db`
+
+"""
+
+
+class RetrieveVariables:
```

Comment:
I think in the future we should make this part a bit better. What we have now with the "old" GCP system tests is not really easy to use: with KMS, retrieving the project etc - it's quite flaky in a way. But for now I think it's good to leave it as it is now.